### PR TITLE
Improve Search Box Speed to make it more "buttery" + type anywhere to search.

### PR DESCRIPTION
**Pull request by wenqinYe**
The live emoji search is great, expect for one problem; it slowed down the system as it made a separate search call for every character typed. This would result in noticeable lag in typing (this is even a problem on my MBP 2012).
So what I did was put a delay on the search so that the search is performed when the user is done typing. This way, there are no useless (and expensive) search calls per character; instead, there is only one call at the end when the user finished typing. 

Of course, one thing that live search was great at doing was giving the feeling of instantaneous search results. I wanted to keep this illusion of "instant" search by making the results disappear to give the user the sense that their search is being instantly acknowledged. Without this, the user is stuck on the previous results page until they finish typing, which can give a feeling of sluggishness from the system.

---

As a side, you can now type anywhere to search; I removed the feature of having to type in a slash first. I don't think people want to type in an arbitrary command when their intent is to access the search bar and find some awesome emojis :smile:. They should be able to access the search bar right away.

Cheers
:beers:

### CODE REVIEW

**Comment by brianchung808** [comment]
You should make the changes in the .coffee files since the js you edited is just generated.

**Comment by wenqinYe** [comment]
Ahh thank you I was not aware that we were supposed to edit the .coffee files.

**Comment by wenqinYe** [comment]
Sorry for the delay but today I switched everything to coffee script.
Could muan please look at this commit; thank you!

Also I decided to keep the "live" search updating instead of everything going away when the user types something and then shows results when the user finishes typing.

Cheers
:beers:

File: _assets/speedy.coffee [hunk 1]
```diff
@@ -30,8 +30,27 @@ setRelatedDOMVisibility = (keyword) ->
...
+delayedSearch = (delayTime)->
+
+  ###
+  We don't want timeouts stacking on top of eachother 
+  which would defeat what the delayed search function does
+  ###
+  clearTimeout(timeout);
+
+  timeout = setTimeout(
+    (()-> search($('.speedy-filter').val()) ), delayTime)
```
Comment (brianchung808) [inline _assets/speedy.coffee]: style-wise it'd be nicer to omit the `()` in the `() ->` and leave it as `->`. (https://github.com/styleguide/javascript)

**Comment by brianchung808** [comment]
The semicolons should be removed (:mask:) but I do like being able to just start typing to search rather than the vim-y `/` method :+1:

**Comment by brianchung808** [comment]
@wenqinYe I downloaded your fork & the `focusOnSearch` doesn't actually focus on the search bar. I sent a pull req with a fix

File: _assets/speedy.coffee (hunk 1 above)
Comment (wenqinYe) [inline _assets/speedy.coffee]: Ok thank you brian :)

**Comment by muan** [comment]
Hai, sorry for the delay in replying! 
@brianchung808 thanks so much for help moving this forward while I be lazy!

> Also I decided to keep the "live" search updating instead of everything going away when the user types something and then shows results when the user finishes typing.

Just pulled it down, looks like the search is not "live"? do you mean that you'd keep the delay but don't clear the previous results? It feels pretty slow to me:



Compared to the current behavior:

**Comment by wenqinYe** [comment]
I originally planned on clearing the previous search results as soon as the user typed in a search request to give the illusion of speed, but I changed it because I thought you might not have wanted that.

**Comment by muan** [comment]
Thanks! I like being able to type/search anywhere.

But regarding the performance issue, given that the lag doesn't exist on every machine(on my MBP it's pretty snappy), the solution here seems to be making search considerably slower for everyone, which isn't great. Have you tried cutting the wait time down to something closer to time between keystrokes?

**Comment by wenqinYe** [comment]
I'm going to work on this again after exams. Been too busy studying right now, will comment back when I have a new commit for you.

**Comment by muan** [comment]
Closing as stale.

**Pull request closed without merging**